### PR TITLE
IN-22 feat: add modal

### PR DESCRIPTION
**Pull request by web-serge**

### CODE REVIEW

File: src/components/modal/modal.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import type { Meta, StoryObj } from "@storybook/react"
+
+import { type ComponentProps, useState } from "react"
+
+import { Button } from "../button"
+import { Card } from "../card"
+import { Icon } from "../icon"
+import { Typography } from "../typography"
```
Comment (Abrosimov96) [inline src/components/modal/modal.stories.tsx:8]:
```
type ComponentProps,
import { Card } from "../card"
import { Icon } from "../icon"
import { Typography } from "../typography"
```
Unused imports

**Review by Abrosimov96 — APPROVED** [review]

File: src/components/modal/modal.stories.tsx (hunk 1 above)
Comment (web-serge) [reply src/components/modal/modal.stories.tsx]: fix

File: src/components/modal/modal.module.scss [hunk 2]
```diff
@@ -0,0 +1,80 @@
+@use "../../styles/abstracts/index" as *;
+
+.dialogOverlay {
+  position: fixed;
+  inset: 0;
+  background-color: var(--color-dark-700);
```
Comment (vitaliiRss) [inline src/components/modal/modal.module.scss:6]: Please double check color, I guess here should be rgba format with some transparency

Comment (web-serge) [reply src/components/modal/modal.module.scss]: change color on "rgb(0 0 0 / 60%)"

**Review by vitaliiRss — APPROVED** [review]

**Pull request merged**